docs/parser_plugins.md

```diff
@@ -1 +1,264 @@
+Note that `myparser.py` and `mypackage.py` just act as a blueprints.
+They should not be edited or exposed directly.
+
+The `README.md` should remain a copy of the template, in case anyone wants to fork the project.
+Instead, place plugin descriptions in the entry points or GitHub repository metadata.
+More elaborate explanations should go under `docs`.
+You can deploy them on GitHub or locally via `mkdocs`.
+Lastly, NOMAD uses the Apache 2.0 license (option 4 in the `cruft` setup).
+Please select the same license for maximal legal compatibility.
+
+```bash
+├── nomad-plugin-parser
+│   ├── src
+|   │   ├── nomad_plugin_parser
+|   |   │   ├── parsers
+|   |   │   │   ├── myparser.py  "(target functionality)"
+|   |   │   │   ├── __init__.py  "(entry point)"
+|   |   │   ├── schema_packages
+|   |   │   │   ├── mypackage.py
+|   |   │   │   ├── __init__.py
+│   ├── docs
+│   ├── tests
+│   ├── pyproject.toml           "(module setup file)"
+│   ├── LICENSE.txt
+│   ├── README.md
+-------------------------------------------------------------
+├── nomad.yaml                   "(NOMAD configuration file)"
+```
+
+This examples also highlights the files containing _entry point_ information between parentheses.
+They are explained in greater detail in section [[Extra: Managing Entry Points]].
+For now, it suffices to understand them as the official mechanism for `pip` installing modules.
+The dependence runs bottom up, i.e. the `module setup file` refers to `entry point` that refers to `target functionality`.
+
+The final entry point piece (`NOMAD configuration file`) resides in `nomad.yaml` and refers back to the `module setup file`.
+Normally, you don't have to touch anything there for NOMAD to pick up on your parser.
+It does allows you to control the loading of plugins and their options.
+Closing the "dependence" cycle, `myparser.py` has to read in these options.
+
+## Fundamentals of Parsing
+
+As parsing involves the mapping between two well-defined formats, one could expect it to be trivial.
+In practice, however, parser developers have to manage discrepancies in semantics, shape, data type or units.
+This has lead to five distinct categories of responsibility for the developer to manage.
+Here, they are ordered to match the parser's execution:
+
+1. **file selection** - navigate the upload's folder structure and select the relevant files.
+2. **source extraction** - read the files into Python. This step may already include some level of data field filtering.
+3. **source to target** - map the data of interest with their counterparts in the target/NOMAD schema. This is where the bulk of the filtering happens.
+4. **data mangling** - manipulate the data to match the target/NOMAD `Quantity`s' specification, e.g. dimensionality, shape. This may include computing derived properties not present in the original source files.
```

Comment:
I think we can drop "target" now and just say NOMAD

"... NOMAD's `Quantity` specification..."